### PR TITLE
Upgrade our .NET SDK to 5.0 RC 1

### PR DESCRIPTION
**Pull request by jasonmalinowski**

### CODE REVIEW

File: eng/targets/Imports.targets [hunk 1]
```diff
@@ -404,4 +404,10 @@
     </ItemGroup>
   </Target>
 
+  <!-- Override this to work around https://github.com/dotnet/sdk/issues/13427. The property doesn't
+       appear to be used for anything other than controlling the warning. -->
+  <PropertyGroup>
+    <_MicrosoftWindowsDesktopSdkImported>false</_MicrosoftWindowsDesktopSdkImported>
+  </PropertyGroup>
```
Comment (sharwell) [inline eng/targets/Imports.targets:411]: 📝 File a bug to fix this when we move to RC2, or submit a draft PR so it's easy to take whenever

**Review by sharwell — APPROVED** [review]

File: src/Compilers/Core/Portable/InternalUtilities/PlatformAttributes.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
```
Comment (jaredpar) [inline src/Compilers/Core/Portable/InternalUtilities/PlatformAttributes.cs:5]: This should be unnecassary. Not blocking though.

**Review by jaredpar — APPROVED** [review]

**Review by 333fred — APPROVED** [review]
Please send a mop-up to update the public build instructions doc, and send an email to roslyn x-team as well.

**Comment by jasonmalinowski** [comment]
PR is created for the documentation update: https://github.com/dotnet/roslyn/pull/48421

**Pull request merged**